### PR TITLE
refactor(p2p): remove unused code & collate code marked for deprecation

### PR DESCRIPTION
**Pull request by ramfox**
Cull unused code and related tests! All code marked for deprecation is now in `p2p_deprecate.go`

### CODE REVIEW

File: p2p/datasets.go [hunk 1]
```diff
@@ -45,14 +48,22 @@ func (n *QriNode) RequestDatasetsList(ctx context.Context, pid peer.ID, p Datase
 
 	req = req.WithHeaders("phase", "request")
 
-	replies := make(chan Message)
-	err = n.SendMessage(ctx, req, replies, pid)
+	s, err := n.host.NewStream(ctx, pid, depQriProtocolID)
 	if err != nil {
-		log.Debug(err.Error())
-		return nil, fmt.Errorf("send dataset info message error: %s", err.Error())
+		return nil, fmt.Errorf("error opening stream: %s", err.Error())
```
Comment (ramfox) [inline p2p/datasets.go:53]: Note: replacing mentions of `n.SendMessage`. It relies on code that is already removed & is way overkill for a situation like this, where all you are doing is directly asking one peer for a response.

File: p2p/datasets_test.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ import (
 )
 
 func TestRequestDatasetsList(t *testing.T) {
```
Comment (ramfox) [inline p2p/datasets_test.go:11]: This test is slated 2nd for refactoring when we overhaul the p2p test network setup!

File: p2p/node.go [hunk 3]
```diff
@@ -63,12 +63,6 @@ type QriNode struct {
 	// localResolver allows the node to resolve local dataset references
 	localResolver dsref.Resolver
 
-	// handlers maps this nodes registered handlers. This works in a way
-	// similary to a router in traditional client/server models, but messages
-	// are flying around all over the place instead of a
-	// request/response pattern
-	handlers map[MsgType]HandlerFunc
```
Comment (ramfox) [inline p2p/node.go:70]: This is only used for the `depQriProtocolID` protocol. All stream handling for the `depQriProtocolID` has been moved to func `depQriStreamHandler` in `p2p_deprecate.go`

File: p2p/node.go [hunk 4]
```diff
@@ -425,12 +371,6 @@ func (n *QriNode) disconnected(_ net.Network, conn net.Conn) {
 	n.qis.HandleQriPeerDisconnect(pi.ID)
 }
 
-// QriStreamHandler is the handler we register with the multistream muxer
-func (n *QriNode) QriStreamHandler(s net.Stream) {
```
Comment (ramfox) [inline p2p/node.go:429]: This func and the below `handleStream` have been combined into the `depQriStreamHandler` func

File: p2p/list_peers_test.go [hunk 5]
```diff
@@ -8,7 +8,7 @@ import (
 )
 
 func TestListPeers(t *testing.T) {
-	t.Skip("ramfox: flakey test until ConnectNodes is refactored")
+	t.Skip("ramfox: flakey test until p2p test network setup is refactored")
```
Comment (ramfox) [inline p2p/list_peers_test.go:11]: This test is also slated for refactoring when we change the p2p network setup.

**Review by b5 — APPROVED** [review]
Looks good to me!

**Pull request merged**